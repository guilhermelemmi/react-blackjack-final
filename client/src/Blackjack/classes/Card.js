import { RANKS_MAP, SUITS_MAP } from "../constants";

class Card {
  suit;
  suitName;
  rank;
  rankName;

  constructor(suit, rank) {
    this.suit = suit;
    this.rank = rank;
    this.suitName = SUITS_MAP[suit];
    this.rankName = RANKS_MAP[rank];
  }
}

export default Card;
