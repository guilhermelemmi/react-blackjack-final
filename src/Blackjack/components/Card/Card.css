.card {
  display: block;
  height: 291px;
  width: 200px;
  background: url(../../assets/cards.webp) no-repeat 0 0;
  border: 3px solid #fff;
  border-radius: 8px;
  box-shadow: 2px 2px 12px #333;
}

.closed {
  background: url(../../assets/card-back.webp) repeat 0 0;
  background-size: contain;
}

._132 {
  background-position: -200px 0;
}

._133 {
  background-position:-400px 0;
}

._134 {
  background-position: -600px 0;
}

._121 {
  background-position: 0 -290px;
}

._122 {
  background-position: -200px -290px;
}

._123 {
  background-position: -400px -290px;
}

._124 {
  background-position: -600px -290px;
}

._111 {
  background-position: 0 -581px;
}

._112 {
  background-position: -200px -581px;
}

._113 {
  background-position: -400px -581px;
}

._114 {
  background-position: -600px -581px;
}

._101 {
  background-position: 0 -871px;
}

._102 {
  background-position: -200px -871px;
}

._103 {
  background-position: -400px -871px;
}

._104 {
  background-position: -600px -871px;
}

._91 {
  background-position: 0 -1162px;
}

._92 {
  background-position: -200px -1162px;
}

._93 {
  background-position: -400px -1162px;
}

._94 {
  background-position: -600px -1162px;
}

._81 {
  background-position: 0 -1452px;
}

._82 {
  background-position: -200px -1452px;
}

._83 {
  background-position: -400px -1452px;
}

._84 {
  background-position: -600px -1452px;
}

._71 {
  background-position: 0 -1743px;
}

._72 {
  background-position: -200px -1743px;
}

._73 {
  background-position: -400px -1743px;
}

._74 {
  background-position: -600px -1743px;
}

._61 {
  background-position: 0 -2033px;
}

._62 {
  background-position: -200px -2033px;
}

._63 {
  background-position: -400px -2033px;
}

._64 {
  background-position: -600px -2033px;
}

._51 {
  background-position: 0 -2324px;
}

._52 {
  background-position: -200px -2324px;
}

._53 {
  background-position: -400px -2324px;
}

._54 {
  background-position: -600px -2324px;
}

._41 {
  background-position: 0 -2615px;
}

._42 {
  background-position: -200px -2615px;
}

._43 {
  background-position: -400px -2615px;
}

._44 {
  background-position: -600px -2615px;
}

._31 {
  background-position: 0 -2905px;
}

._32 {
  background-position: -200px -2905px;
}

._33 {
  background-position: -400px -2905px;
}

._34 {
  background-position: -600px -2905px;
}

._21 {
  background-position: 0 -3196px;
}

._22 {
  background-position: -200px -3196px;
}

._23 {
  background-position: -400px -3196px;
}

._24 {
  background-position: -600px -3196px;
}

._11 {
  background-position: 0 -3486px;
}

._12 {
  background-position: -200px -3486px;
}

._13 {
  background-position: -400px -3486px;
}

._14 {
  background-position: -600px -3486px;
} 