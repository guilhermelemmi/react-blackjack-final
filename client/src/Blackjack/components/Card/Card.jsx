import { RANKS_MAP, SUITS_MAP } from "../../constants";
import "./Card.css";

function Card ({ rank, suit, isClosed = false }) {
  return  (
    <div
      role="img"
      className={isClosed ? 'card closed' : `card _${rank}${suit}`}
      aria-label={isClosed ? 'Hole Card' : `${RANKS_MAP[rank]} of ${SUITS_MAP[suit]}`}
    />
  )
}

export default Card;