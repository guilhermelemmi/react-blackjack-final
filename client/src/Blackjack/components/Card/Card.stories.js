import Card from "./Card";

export default {
  component: Card,
};

export const AceOfSpades = {
  render: () => <Card suit={1} rank={1} />,
};

export const AceOfClubs = {
  render: () => <Card suit={2} rank={1} />,
};

export const AceOfHearts = {
  render: () => <Card suit={3} rank={1} />,
};

export const AceOfDiamonds = {
  render: () => <Card suit={4} rank={1} />,
};

export const CardClosed = {
  render: () => <Card suit={4} rank={1} isClosed={true} />,
};
