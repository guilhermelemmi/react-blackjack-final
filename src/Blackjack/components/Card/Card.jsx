import { RANKS_MAP, SUITS_MAP } from "../../constants";
import "./Card.css";

function Card ({ rank, suit, isClosed = false }) {
  return  (
    <div>
      <span
        role="img"
        className={isClosed ? 'card closed' : `card _${rank}${suit}`}
        aria-label={isClosed ? 'Hole Card' : `${RANKS_MAP[rank]} of ${SUITS_MAP[suit]}`} />
    </div>
  )
}

export default Card;