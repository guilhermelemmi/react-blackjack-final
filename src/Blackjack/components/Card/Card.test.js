import { render } from "@testing-library/react";
import Card from "./Card";

describe("Card", () => {
  test("Card opens matches snapshot", async () => {
    const view = render(<Card suit={1} rank={1} />);
    expect(view.container).toMatchSnapshot();
  });

  test("Card closed matches snapshot", async () => {
    const view = render(<Card suit={1} rank={1} isClosed={true} />);
    expect(view.container).toMatchSnapshot();
  });
});
