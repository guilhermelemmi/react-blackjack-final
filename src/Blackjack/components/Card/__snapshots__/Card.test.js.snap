// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card closed matches snapshot 1`] = `
<div>
  <div>
    <span
      aria-label="Hole Card"
      class="card closed"
      role="img"
    />
  </div>
</div>
`;

exports[`Card Card opens matches snapshot 1`] = `
<div>
  <div>
    <span
      aria-label="Ace of Spades"
      class="card _11"
      role="img"
    />
  </div>
</div>
`;
